spec/support/shared_examples/validators/shared_examples_for_advocate_claim_validators.rb

```diff
@@ -1,53 +1,121 @@
-RSpec.shared_examples 'advocate category validations' do |options|
-  let(:claim) { create(options[:factory]) }
+RSpec.shared_examples 'default valid advocate categories' do |options|
+  context "with factory='#{options[:factory]}', trait='#{options[:trait]}' and form_step='#{options[:form_step]}'" do
+    subject(:claim) { create(options[:factory], options[:trait]) }
 
-  before do
-    claim.form_step = options[:form_step]
-  end
+    before { claim.form_step = options[:form_step] }
+
+    it 'is valid for QC' do
+      claim.advocate_category = 'QC'
+      should_not_error(claim, :advocate_category)
```

Comment:
The `should_not_error` method calls `claim.valid?` and then checks that `claim.errors[:advocate_category]` is empty (similarly in `should_error_with`). I would think that it is sufficient just to test `claim.valid?`. These could tests could then be replaced with:

```ruby
context 'with QC advocate category' do
  before { claim.advocate_category = 'QC' }
  it { is_expected.to be_valid }
end
(etc)
```

How do people feel about this?

Reply:
The `should_not_error` method is expliclty testing if `claim.errors[:advocate_category]` is populated, regardless of whether the claim is valid or not. This is, from memory, because the claim may not be valid (due to factory issues and claim creation setup. It was done like this to enable testing the validity of a specific field, rather than the object itself. 

Even if all tests pass now, this change would mean that changes to factories or setup unrelated to these tests could cause them to break, which i believe is why it was done this way

Reply:
Consider closing if no-one has a chance to review in the next few days